rpc/utils.go

```diff
@@ -218,29 +218,27 @@ func newSubscriptionId() (string, error) {
 // SupportedModules returns the collection of API's that the RPC server offers
 // on which the given client connects.
 func SupportedModules(client Client) (map[string]string, error) {
-	req := map[string]interface{}{
-		"id":     1,
-		"method": "rpc_modules",
+	req := JSONRequest{
+		Id:      new(int64),
+		Version: "2.0",
+		Method:  "rpc_modules",
 	}
-
 	if err := client.Send(req); err != nil {
 		return nil, err
 	}
 
-	var response map[string]interface{}
-	if err := client.Recv(&response); err != nil {
+	response := new(JSONSuccessResponse)
```

Comment:
Actually, you were right that `new` moves it to the heap:

```
go build --gcflags -m 2>&1 | grep utils.go

./utils.go:230: new(JSONSuccessResponse) escapes to heap
```

Unfortunately if you take the address of a struct then it also immediately escapes to the heap. With your version:

```
go build --gcflags -m 2>&1 | grep utils.go

./utils.go:231: &response escapes to heap
./utils.go:230: moved to heap: response
./utils.go:231: &response escapes to heap
```